Source/Core/DistanceDisplayCondition.js

```diff
@@ -0,0 +1,136 @@
+        near = defaultValue(near, 0.0);
+        this._near = near;
+        this._near2 = near * near;
+
+        far = defaultValue(far, Number.MAX_VALUE);
+        this._far = far;
+        this._far2 = far * far;
+    }
+
+    defineProperties(DistanceDisplayCondition.prototype, {
+        /**
+         * The smallest distance in the interval where the object is visible.
+         * @memberof DistanceDisplayCondition.prototype
+         * @type {Number}
+         * @default 0.0
+         */
+        near : {
+            get : function() {
+                return this._near;
+            },
+            set : function(value) {
+                this._near = value;
+                this._near2 = value * value;
+            }
+        },
+        /**
+         * The largest distance in the interval where the object is visible.
+         * @memberof DistanceDisplayCondition.prototype
+         * @type {Number}
+         * @default Number.MAX_VALUE
+         */
+        far : {
+            get : function() {
+                return this._far;
+            },
+            set : function(value) {
+                this._far = value;
+                this._far2 = value * value;
+            }
+        }
+    });
+
+    var scratchPosition = new Cartesian3();
+
+    /**
+     * Determines is a model is visible based on the current frame state.
+     * @private
+     *
+     * @param {Model} model The model.
+     * @param {FrameState} frameState The current frame state.
+     * @return {Boolean} Whether the model is visible.
+     */
+    DistanceDisplayCondition.prototype.isVisible = function(model, frameState) {
+        // Distance to center of primitive's reference frame
+        var position = Matrix4.getTranslation(model.modelMatrix, scratchPosition);
+        var distance2 = Cartesian3.distanceSquared(position, frameState.camera.positionWC);
+        return (distance2 >= this._near2) && (distance2 <= this._far2);
+    };
+
+    /**
+     * Determines if two distance display conditions are equal.
+     *
+     * @param {DistanceDisplayCondition} left A distance display condition.
+     * @param {DistanceDisplayCondition} right Another distance display condition.
+     * @return {Boolean} Whether the two distance display conditions are equal.
+     */
+    DistanceDisplayCondition.equals = function(left, right) {
+        return left === right ||
+               (defined(left) &&
+                defined(right) &&
+                left.near === right.near &&
+                left.far === right.far);
+    };
+
+    /**
+     * Duplicates a distance display condition instance.
+     *
+     * @param {DistanceDisplayCondition} [value] The distance display condition to duplicate.
+     * @param {DistanceDisplayCondition} [result] The result onto which to store the result.
+     * @return {DistanceDisplayCondition} The duplicated instance.
+     */
+    DistanceDisplayCondition.clone = function(value, result) {
```

Comment:
Since this is used directly by the Entity api as well, it should have a prototype clone too.  Right now the Entity API ends up making a new instance every time get value is called because no prototype clone is defined.